src/app-admin.js

```diff
@@ -0,0 +1,491 @@
+    ) || {};
+
+  res.json(cnt > 0);
+});
+
+app.get('/users/', (req, res) => {
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  const users = getAccountDb().all(
+    `SELECT users.id, user_name as userName, display_name as displayName, enabled, ifnull(master,0) as master, roles.id as role 
+     FROM users
+     JOIN user_roles ON user_roles.user_id = users.id
+     JOIN roles ON roles.id = user_roles.role_id
+     WHERE users.user_name <> ''`,
+  );
+
+  res.json(
+    users.map((u) => ({
+      ...u,
+      master: u.master === 1,
+      enabled: u.enabled === 1,
+    })),
+  );
+});
+
+app.post('/users', async (req, res) => {
+  const user = await getAdminSessionFromRequest(req, res);
+  if (!user) return;
+
+  const newUser = req.body;
+  const { id: userIdInDb } = getUserByUsername(newUser.userName);
+
+  if (!validateUserInput(res, newUser)) return;
+  if (userIdInDb) {
+    sendErrorResponse(
+      res,
+      400,
+      'user-already-exists',
+      `User ${newUser.userName} already exists`,
+    );
+    return;
+  }
+
+  const userId = uuid.v4();
+  let displayName = newUser.displayName || null;
+  let enabled = newUser.enabled ? 1 : 0;
+
+  getAccountDb().mutate(
+    'INSERT INTO users (id, user_name, display_name, enabled, master) VALUES (?, ?, ?, ?, 0)',
+    [userId, newUser.userName, displayName, enabled],
+  );
+
+  getAccountDb().mutate(
+    'INSERT INTO user_roles (user_id, role_id) VALUES (?, ?)',
+    [userId, newUser.role],
+  );
+
+  res.status(200).send({ status: 'ok', data: { id: userId } });
+});
+
+app.patch('/users', async (req, res) => {
+  const user = await getAdminSessionFromRequest(req, res);
+  if (!user) return;
+
+  const userToUpdate = req.body;
+  const { id: userIdInDb } =
+    getAccountDb().first('SELECT id FROM users WHERE id = ?', [
+      userToUpdate.id,
+    ]) || {};
+
+  if (!validateUserInput(res, userToUpdate)) return;
+  if (!userIdInDb) {
+    sendErrorResponse(
+      res,
+      400,
+      'cannot-find-user-to-update',
+      `Cannot find ${userToUpdate.userName} to update`,
+    );
+    return;
+  }
+
+  let displayName = userToUpdate.displayName || null;
+  let enabled = userToUpdate.enabled ? 1 : 0;
+
+  getAccountDb().mutate(
+    'UPDATE users SET user_name = ?, display_name = ?, enabled = ? WHERE id = ?',
+    [userToUpdate.userName, displayName, enabled, userIdInDb],
+  );
+
+  getAccountDb().mutate('UPDATE user_roles SET role_id = ? WHERE user_id = ?', [
+    userToUpdate.role,
+    userIdInDb,
+  ]);
+
+  res.status(200).send({ status: 'ok', data: { id: userIdInDb } });
+});
+
+app.post('/users/delete-all', async (req, res) => {
+  const user = await getAdminSessionFromRequest(req, res);
+  if (!user) return;
+
+  const ids = req.body.ids;
+  let totalDeleted = 0;
+  ids.forEach((item) => {
+    const { id: masterId } =
+      getAccountDb().first('SELECT id FROM users WHERE master = 1') || {};
+
+    if (item === masterId) return;
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏missing HTTP response (400)